tracing-subscriber/src/fmt/fmt_subscriber.rs

```diff
@@ -1221,20 +1212,87 @@ where
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 struct Timings {
-    idle: u64,
-    busy: u64,
+    idle: Duration,
+    busy: Duration,
     last: Instant,
+    entered_count: u64,
 }
 
+/// Tracks the idle and busy times of a span.
+///
+/// The number of times the span has been entered is tracked to determine the current state. This
+/// ensures that entering and exiting the span multiple times out of order does not affect the
+/// timings.
 impl Timings {
     fn new() -> Self {
         Self {
-            idle: 0,
-            busy: 0,
+            // The total time spent in the idle state not including the current idle time.
+            idle: Duration::ZERO,
+            // The total time spent in the busy state not including the current busy time.
+            busy: Duration::ZERO,
+            // The last time the state was changed.
             last: Instant::now(),
+            // The number of times the busy state has been entered. This is used to track the
+            // current state.
```

Comment:
Why not put these comments as doc comments on the fields themselves? Especially since it could hopefully one day go public.

Reply:
Clippy complains about doc comments on private fields.

Reply:
On second though the fields will probably not be public anyway, so that users can't manually change them, right?

I'd say this is fine for now either way.

Reply:
It's easy enough to change if needed, so no need to worry about this until this gets made public.